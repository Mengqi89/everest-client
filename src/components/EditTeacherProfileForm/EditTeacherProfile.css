.edit-teacher {
    display: flex;
    flex-direction: column;
}