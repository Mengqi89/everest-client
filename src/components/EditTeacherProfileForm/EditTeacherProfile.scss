@import '../../assets/variables.scss';


.edit-teacher {
    display: flex;
    flex-direction: column;
    margin: auto;
    width: 30%;

    input,
    textarea,
    select {
      @include input-big;
    }

    input,
    select,
    textarea {
      margin-top: 0.5em;
    }

    select {
        padding: 0;
    }

    label {
        margin-top: 1em;
    }

    button {
        @include purple-button;
        width: 200px;
      }
}