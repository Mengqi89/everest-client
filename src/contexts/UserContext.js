import React, { useState } from 'react'

export const nullSchoolUser = {
    housingInformation: {},
    insuranceInformation: {},
    visaInformation: {},
    otherServices: {},
    lifestyleInformation: {}
}

const UserContext = React.createContext({
    loggedIn: null,
    setLoggedIn: () => { },
    userType: '',
    setUserType: () => { },
    user: {},
    setUser: () => { },
    applications: [],
    setApplications: () => { },
    error: null,
    setError: () => { },
})

export default UserContext

export function UserProvider(props) {
    const [loggedIn, setLoggedIn] = useState(null)
    const [userType, setUserType] = useState('')
    const [user, setUser] = useState(nullSchoolUser)
    const [applications, setApplications] = useState([])
    const [error, setError] = useState(null)

    const setLoggedInFn = loggedIn => {
        setLoggedIn(loggedIn)
    }

    const setUserTypeFn = userType => {
        setUserType(userType)
    }

    const setUserFn = user => {
        setUser(user)
    }

    const setApplicationsFn = applications => {
        setApplications(applications)
    }

    const setErrorFn = error => {
        setError(error)
    }


    const value = {
        loggedIn: loggedIn,
        setLoggedIn: setLoggedInFn,
        userType: userType,
        setUserType: setUserTypeFn,
        user: user,
        setUser: setUserFn,
        applications: applications,
        setApplications: setApplicationsFn,
        error: error,
        setError: setErrorFn
    }

    return (
        <UserContext.Provider value={value}>
            {props.children}
        </UserContext.Provider>
    )
}