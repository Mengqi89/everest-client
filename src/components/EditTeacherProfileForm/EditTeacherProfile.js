import React, { Component } from 'react'
import NationalityList from '../TeacherRegistrationForm/NationalityList'
import './EditTeacherProfile.scss'
import TeacherApiService from '../../services/teacher-api-service'
import UserContext from '../../contexts/UserContext'
import { withRouter } from 'react-router-dom'

class EditTeacherProfile extends Component {
    static contextType = UserContext

    componentDidMount() {
        this.context.setError(null)
        TeacherApiService.getTeacherProfile()
            .then(teacher => {
                this.context.setUser(teacher)
            })
            .catch(err => console.log(err))
    }

    onFormSubmit = (ev) => {
        ev.preventDefault()
        const { id } = this.context.user

        const { username, password, first_name, last_name, age, sex, race, nationality, native_speaker,
            married, highest_degree, field_of_degree, school, certification, years_of_experience,
            years_in_china, years_teaching_abroad } = ev.target

        const updatedTeacher = {
            username: username.value,
            password: password.value,
            first_name: first_name.value,
            last_name: last_name.value,
            age: age.value,
            sex: sex.value,
            race: race.value,
            nationality: nationality.value,
            native_speaker: native_speaker.value,
            married: married.value,
            highest_degree: highest_degree.value,
            field_of_degree: field_of_degree.value,
            school: school.value,
            certification: certification.value,
            years_of_experience: years_of_experience.value,
            years_in_china: years_in_china.value,
            years_teaching_abroad: years_teaching_abroad.value
        }

        TeacherApiService.updateTeacherProfile(id, updatedTeacher)
            .then(res => this.props.history.push('/profile'))
            .catch(err => this.context.setError(err))
    }

    render() {
        const { user, error } = this.context
        {console.log(user)}
        return (
            <div>
            <h1>Edit Teacher Profile</h1>
                
                <form className='form edit-teacher' onSubmit={(ev) => this.onFormSubmit(ev)}>
                    <div role='alert'>
                        {error && <p className='red'>{error}</p>}
                    </div>

                    <label htmlFor='teacher-username'>Enter a Username</label>
                    <input id='teacher-usernmae' type='text' name='username' defaultValue={user.username} required></input>

                    <label htmlFor='teacher-password'>Enter a Password</label>
                    <input id='teacher-password' type='password' 
                    title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more 
                    characters" name='password' placeholder='please enter password' required></input>

                    <label htmlFor='teacher-firstname'>First Name</label>
                    <input id='teacher-firstname' type='text' name='first_name' defaultValue={user.first_name} required></input>

                    <label htmlFor='teacher-lastname'>Last Name</label>
                    <input id='teacher-lastname' type='text' name='last_name' defaultValue={user.last_name} required></input>

                    <label htmlFor='teacher-age'>Age</label>
                    <input id='teacher-age' type='number' name='age' defaultValue={user.age} required></input>
                            
                    <label htmlFor='teacher-sex'>Gender</label>
                    <select id='teacher-sex' name='sex' defaultValue={user.sex} required>
                        <option value='Male'>Male</option>
                        <option value='Female'>Female</option>
                        <option value='Gender Neutral'>Gender-Neutral</option>
                    </select>

                    <label htmlFor='teacher-nationality'>Nationality</label>
                    <NationalityList id='teacher-nationality' nationality={user.nationality}/>

                    <label htmlFor='race'>Ethnicity</label>
                    <select id='race' name='race' defaultValue={user.race} required>
                        <option value="" disabled>--select one--</option>
                        <option value='African American'>African American</option>
                        <option value='Asian'>Asian</option>
                        <option value='Native American'>Native American</option>
                        <option value='Latino'>Latino</option>
                        <option value='Pacific Islander'>Pacific Islander</option>
                        <option value='Multiracial'>Multiracial</option>
                        <option value='White'>White</option>
                        <option value='Other'>Other</option>
                    </select>

                    <label htmlFor='teacher-native-speaker'>Are You A Native Speaker?</label>
                    <select id='teacher-native-speaker' name='native_speaker' defaultValue={user.native_speaker} required>
                        <option value='true'>Yes</option>
                        <option value='false'>No</option>
                    </select>

                    <label htmlFor='teacher-married'>Marraige Satus</label>
                    <select id='teacher-married' name='married' defaultValue={user.married} required>
                        <option value="" disabled>--select one--</option>
                        <option value='Married'>married</option>
                        <option value='Single'>single</option>
                    </select>

                    <label htmlFor='teacher-education'>Education</label>
                    <select id="teacher-education" name="highest_degree" defaultValue={user.highest_degree} required>
                        <option value="" disabled="disabled">-- select one --</option>
                        <option value="No formal education">No formal education</option>
                        <option value="Primary education">Primary education</option>
                        <option value="Secondary education">Secondary education or high school</option>
                        <option value="GED">GED</option>
                        <option value="Vocational qualification">Vocational qualification</option>
                        <option value="BA">Bachelor's degree</option>
                        <option value="Master's degree">Master's degree</option>
                        <option value="Doctorate or higher">Doctorate or higher</option>
                    </select>

                    <label htmlFor='teacher-degree'>Degree</label>
                    <select id='teacher-degree' name='field_of_degree' defaultValue={user.field_of_degree} required>
                        <option value='No Degree'>No Degree</option>
                        <option value='English'>English</option>
                        <option value='Business'>Business</option>
                        <option value='Marketing'>Marketing</option>
                        <option value='Accounting'>Accounting</option>
                        <option value='Other'>Other</option>
                    </select>

                    <label htmlFor='teacher-school'>School Attended</label>
                    <input id='teacher-school' type='text' name='school' defaultValue={user.school}required></input>

                    <label htmlFor='teacher-certification'>Certification</label>
                    <select id='teacher-certification' name='certification' defaultValue="None" required>
                        <option value='None'>None</option>
                        <option value='TEFL'>TEFL</option>
                        <option value='TESOL'>TESOL</option>
                        <option value='TEFL/TESOL'>TEFL/TESOL</option>
                    </select>

                    <label htmlFor='teacher-experience'>Years Of Experience</label>
                    <select id='teacher-experience' name='years_of_experience' defaultValue={user.years_of_experience} required>
                        <option value='0'>0</option>
                        <option value='1'>1</option>
                        <option value='2'>2</option>
                        <option value='3'>3+</option>
                    </select>

                    <label htmlFor='teacher-experience-inChina'>Years Of Experience Teaching In China</label>
                    <select id='teacher-experience-inChina' name='years_in_china' defaultValue={user.years_in_china} required>
                        <option value='0'>0</option>
                        <option value='1'>1</option>
                        <option value='2'>2</option>
                        <option value='3'>3+</option>
                    </select>

                    <label htmlFor='teacher-experience-notChina'>Years Of Experience Teaching - Other Countries</label>
                    <select id='teacher-experience-notChina' name='years_teaching_abroad' defaultValue={user.years_teaching_abroad} required>
                        <option value='0'>0</option>
                        <option value='1'>1</option>
                        <option value='2'>2</option>
                        <option value='3'>3+</option>
                    </select>

                    <button type='submit'>Submit</button>
                </form>
            </div>
        )
    }
}

export default withRouter(EditTeacherProfile)