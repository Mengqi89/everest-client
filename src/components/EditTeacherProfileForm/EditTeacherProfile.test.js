import React from 'react'
import EditTeacherProfile from './EditTeacherProfile'
import teachers from '../../dummyData/teacher'
import { BrowserRouter } from 'react-router-dom'
import UserContext from '../../contexts/UserContext'
import { mount } from 'enzyme';


describe('EditTeacherProfile', () => {
  it('should render correctly', () => {
    const component = mount(<BrowserRouter>
      <UserContext.Provider value={{ user: teachers[0] }}>
        <EditTeacherProfile />
      </UserContext.Provider>
    </BrowserRouter>);

    expect(component).toMatchSnapshot();
  });
});
