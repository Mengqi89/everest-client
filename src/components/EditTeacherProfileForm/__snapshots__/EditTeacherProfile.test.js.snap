// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTeacherProfile should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(EditTeacherProfile)>
      <EditTeacherProfile
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="edit-teacher-form-container"
        >
          <h2>
            Edit Teacher Profile
          </h2>
          <form
            className="form edit-teacher"
            onSubmit={[Function]}
          >
            <div
              role="alert"
            />
            <label
              htmlFor="teacher-username"
            >
              Enter a Username
            </label>
            <input
              defaultValue="NatBowie"
              id="teacher-username"
              name="username"
              required={true}
              type="text"
            />
            <label
              htmlFor="teacher-password"
            >
              Enter a Password
            </label>
            <input
              id="teacher-password"
              name="password"
              placeholder="please enter password"
              required={true}
              title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more  characters"
              type="password"
            />
            <label
              htmlFor="teacher-firstname"
            >
              First Name
            </label>
            <input
              id="teacher-firstname"
              name="first_name"
              required={true}
              type="text"
            />
            <label
              htmlFor="teacher-lastname"
            >
              Last Name
            </label>
            <input
              id="teacher-lastname"
              name="last_name"
              required={true}
              type="text"
            />
            <label
              htmlFor="teacher-age"
            >
              Age
            </label>
            <input
              id="teacher-age"
              min="0"
              name="age"
              required={true}
              type="number"
            />
            <label
              htmlFor="teacher-sex"
            >
              Gender
            </label>
            <select
              id="teacher-sex"
              name="sex"
              required={true}
            >
              <option
                value="Male"
              >
                Male
              </option>
              <option
                value="Female"
              >
                Female
              </option>
              <option
                value="Others"
              >
                Others
              </option>
            </select>
            <label
              htmlFor="teacher-nationality"
            >
              Nationality
            </label>
            <NationalityList
              id="teacher-nationality"
            >
              <select
                id="teacher-nationality"
                name="nationality"
                required={true}
              >
                <option
                  value=""
                >
                  -- select one --
                </option>
                <option
                  value="afghan"
                >
                  Afghan
                </option>
                <option
                  value="albanian"
                >
                  Albanian
                </option>
                <option
                  value="algerian"
                >
                  Algerian
                </option>
                <option
                  value="american"
                >
                  American
                </option>
                <option
                  value="andorran"
                >
                  Andorran
                </option>
                <option
                  value="angolan"
                >
                  Angolan
                </option>
                <option
                  value="antiguans"
                >
                  Antiguans
                </option>
                <option
                  value="argentinean"
                >
                  Argentinean
                </option>
                <option
                  value="armenian"
                >
                  Armenian
                </option>
                <option
                  value="australian"
                >
                  Australian
                </option>
                <option
                  value="austrian"
                >
                  Austrian
                </option>
                <option
                  value="azerbaijani"
                >
                  Azerbaijani
                </option>
                <option
                  value="bahamian"
                >
                  Bahamian
                </option>
                <option
                  value="bahraini"
                >
                  Bahraini
                </option>
                <option
                  value="bangladeshi"
                >
                  Bangladeshi
                </option>
                <option
                  value="barbadian"
                >
                  Barbadian
                </option>
                <option
                  value="barbudans"
                >
                  Barbudans
                </option>
                <option
                  value="batswana"
                >
                  Batswana
                </option>
                <option
                  value="belarusian"
                >
                  Belarusian
                </option>
                <option
                  value="belgian"
                >
                  Belgian
                </option>
                <option
                  value="belizean"
                >
                  Belizean
                </option>
                <option
                  value="beninese"
                >
                  Beninese
                </option>
                <option
                  value="bhutanese"
                >
                  Bhutanese
                </option>
                <option
                  value="bolivian"
                >
                  Bolivian
                </option>
                <option
                  value="bosnian"
                >
                  Bosnian
                </option>
                <option
                  value="brazilian"
                >
                  Brazilian
                </option>
                <option
                  value="british"
                >
                  British
                </option>
                <option
                  value="bruneian"
                >
                  Bruneian
                </option>
                <option
                  value="bulgarian"
                >
                  Bulgarian
                </option>
                <option
                  value="burkinabe"
                >
                  Burkinabe
                </option>
                <option
                  value="burmese"
                >
                  Burmese
                </option>
                <option
                  value="burundian"
                >
                  Burundian
                </option>
                <option
                  value="cambodian"
                >
                  Cambodian
                </option>
                <option
                  value="cameroonian"
                >
                  Cameroonian
                </option>
                <option
                  value="canadian"
                >
                  Canadian
                </option>
                <option
                  value="cape verdean"
                >
                  Cape Verdean
                </option>
                <option
                  value="central african"
                >
                  Central African
                </option>
                <option
                  value="chadian"
                >
                  Chadian
                </option>
                <option
                  value="chilean"
                >
                  Chilean
                </option>
                <option
                  value="chinese"
                >
                  Chinese
                </option>
                <option
                  value="colombian"
                >
                  Colombian
                </option>
                <option
                  value="comoran"
                >
                  Comoran
                </option>
                <option
                  value="congolese"
                >
                  Congolese
                </option>
                <option
                  value="costa rican"
                >
                  Costa Rican
                </option>
                <option
                  value="croatian"
                >
                  Croatian
                </option>
                <option
                  value="cuban"
                >
                  Cuban
                </option>
                <option
                  value="cypriot"
                >
                  Cypriot
                </option>
                <option
                  value="czech"
                >
                  Czech
                </option>
                <option
                  value="danish"
                >
                  Danish
                </option>
                <option
                  value="djibouti"
                >
                  Djibouti
                </option>
                <option
                  value="dominican"
                >
                  Dominican
                </option>
                <option
                  value="dutch"
                >
                  Dutch
                </option>
                <option
                  value="east timorese"
                >
                  East Timorese
                </option>
                <option
                  value="ecuadorean"
                >
                  Ecuadorean
                </option>
                <option
                  value="egyptian"
                >
                  Egyptian
                </option>
                <option
                  value="emirian"
                >
                  Emirian
                </option>
                <option
                  value="equatorial guinean"
                >
                  Equatorial Guinean
                </option>
                <option
                  value="eritrean"
                >
                  Eritrean
                </option>
                <option
                  value="estonian"
                >
                  Estonian
                </option>
                <option
                  value="ethiopian"
                >
                  Ethiopian
                </option>
                <option
                  value="fijian"
                >
                  Fijian
                </option>
                <option
                  value="filipino"
                >
                  Filipino
                </option>
                <option
                  value="finnish"
                >
                  Finnish
                </option>
                <option
                  value="french"
                >
                  French
                </option>
                <option
                  value="gabonese"
                >
                  Gabonese
                </option>
                <option
                  value="gambian"
                >
                  Gambian
                </option>
                <option
                  value="georgian"
                >
                  Georgian
                </option>
                <option
                  value="german"
                >
                  German
                </option>
                <option
                  value="ghanaian"
                >
                  Ghanaian
                </option>
                <option
                  value="greek"
                >
                  Greek
                </option>
                <option
                  value="grenadian"
                >
                  Grenadian
                </option>
                <option
                  value="guatemalan"
                >
                  Guatemalan
                </option>
                <option
                  value="guinea-bissauan"
                >
                  Guinea-Bissauan
                </option>
                <option
                  value="guinean"
                >
                  Guinean
                </option>
                <option
                  value="guyanese"
                >
                  Guyanese
                </option>
                <option
                  value="haitian"
                >
                  Haitian
                </option>
                <option
                  value="herzegovinian"
                >
                  Herzegovinian
                </option>
                <option
                  value="honduran"
                >
                  Honduran
                </option>
                <option
                  value="hungarian"
                >
                  Hungarian
                </option>
                <option
                  value="icelander"
                >
                  Icelander
                </option>
                <option
                  value="indian"
                >
                  Indian
                </option>
                <option
                  value="indonesian"
                >
                  Indonesian
                </option>
                <option
                  value="iranian"
                >
                  Iranian
                </option>
                <option
                  value="iraqi"
                >
                  Iraqi
                </option>
                <option
                  value="irish"
                >
                  Irish
                </option>
                <option
                  value="israeli"
                >
                  Israeli
                </option>
                <option
                  value="italian"
                >
                  Italian
                </option>
                <option
                  value="ivorian"
                >
                  Ivorian
                </option>
                <option
                  value="jamaican"
                >
                  Jamaican
                </option>
                <option
                  value="japanese"
                >
                  Japanese
                </option>
                <option
                  value="jordanian"
                >
                  Jordanian
                </option>
                <option
                  value="kazakhstani"
                >
                  Kazakhstani
                </option>
                <option
                  value="kenyan"
                >
                  Kenyan
                </option>
                <option
                  value="kittian and nevisian"
                >
                  Kittian and Nevisian
                </option>
                <option
                  value="kuwaiti"
                >
                  Kuwaiti
                </option>
                <option
                  value="kyrgyz"
                >
                  Kyrgyz
                </option>
                <option
                  value="laotian"
                >
                  Laotian
                </option>
                <option
                  value="latvian"
                >
                  Latvian
                </option>
                <option
                  value="lebanese"
                >
                  Lebanese
                </option>
                <option
                  value="liberian"
                >
                  Liberian
                </option>
                <option
                  value="libyan"
                >
                  Libyan
                </option>
                <option
                  value="liechtensteiner"
                >
                  Liechtensteiner
                </option>
                <option
                  value="lithuanian"
                >
                  Lithuanian
                </option>
                <option
                  value="luxembourger"
                >
                  Luxembourger
                </option>
                <option
                  value="macedonian"
                >
                  Macedonian
                </option>
                <option
                  value="malagasy"
                >
                  Malagasy
                </option>
                <option
                  value="malawian"
                >
                  Malawian
                </option>
                <option
                  value="malaysian"
                >
                  Malaysian
                </option>
                <option
                  value="maldivan"
                >
                  Maldivan
                </option>
                <option
                  value="malian"
                >
                  Malian
                </option>
                <option
                  value="maltese"
                >
                  Maltese
                </option>
                <option
                  value="marshallese"
                >
                  Marshallese
                </option>
                <option
                  value="mauritanian"
                >
                  Mauritanian
                </option>
                <option
                  value="mauritian"
                >
                  Mauritian
                </option>
                <option
                  value="mexican"
                >
                  Mexican
                </option>
                <option
                  value="micronesian"
                >
                  Micronesian
                </option>
                <option
                  value="moldovan"
                >
                  Moldovan
                </option>
                <option
                  value="monacan"
                >
                  Monacan
                </option>
                <option
                  value="mongolian"
                >
                  Mongolian
                </option>
                <option
                  value="moroccan"
                >
                  Moroccan
                </option>
                <option
                  value="mosotho"
                >
                  Mosotho
                </option>
                <option
                  value="motswana"
                >
                  Motswana
                </option>
                <option
                  value="mozambican"
                >
                  Mozambican
                </option>
                <option
                  value="namibian"
                >
                  Namibian
                </option>
                <option
                  value="nauruan"
                >
                  Nauruan
                </option>
                <option
                  value="nepalese"
                >
                  Nepalese
                </option>
                <option
                  value="new zealander"
                >
                  New Zealander
                </option>
                <option
                  value="ni-vanuatu"
                >
                  Ni-Vanuatu
                </option>
                <option
                  value="nicaraguan"
                >
                  Nicaraguan
                </option>
                <option
                  value="nigerien"
                >
                  Nigerien
                </option>
                <option
                  value="north korean"
                >
                  North Korean
                </option>
                <option
                  value="northern irish"
                >
                  Northern Irish
                </option>
                <option
                  value="norwegian"
                >
                  Norwegian
                </option>
                <option
                  value="omani"
                >
                  Omani
                </option>
                <option
                  value="pakistani"
                >
                  Pakistani
                </option>
                <option
                  value="palauan"
                >
                  Palauan
                </option>
                <option
                  value="panamanian"
                >
                  Panamanian
                </option>
                <option
                  value="papua new guinean"
                >
                  Papua New Guinean
                </option>
                <option
                  value="paraguayan"
                >
                  Paraguayan
                </option>
                <option
                  value="peruvian"
                >
                  Peruvian
                </option>
                <option
                  value="polish"
                >
                  Polish
                </option>
                <option
                  value="portuguese"
                >
                  Portuguese
                </option>
                <option
                  value="qatari"
                >
                  Qatari
                </option>
                <option
                  value="romanian"
                >
                  Romanian
                </option>
                <option
                  value="russian"
                >
                  Russian
                </option>
                <option
                  value="rwandan"
                >
                  Rwandan
                </option>
                <option
                  value="saint lucian"
                >
                  Saint Lucian
                </option>
                <option
                  value="salvadoran"
                >
                  Salvadoran
                </option>
                <option
                  value="samoan"
                >
                  Samoan
                </option>
                <option
                  value="san marinese"
                >
                  San Marinese
                </option>
                <option
                  value="sao tomean"
                >
                  Sao Tomean
                </option>
                <option
                  value="saudi"
                >
                  Saudi
                </option>
                <option
                  value="scottish"
                >
                  Scottish
                </option>
                <option
                  value="senegalese"
                >
                  Senegalese
                </option>
                <option
                  value="serbian"
                >
                  Serbian
                </option>
                <option
                  value="seychellois"
                >
                  Seychellois
                </option>
                <option
                  value="sierra leonean"
                >
                  Sierra Leonean
                </option>
                <option
                  value="singaporean"
                >
                  Singaporean
                </option>
                <option
                  value="slovakian"
                >
                  Slovakian
                </option>
                <option
                  value="slovenian"
                >
                  Slovenian
                </option>
                <option
                  value="solomon islander"
                >
                  Solomon Islander
                </option>
                <option
                  value="somali"
                >
                  Somali
                </option>
                <option
                  value="south african"
                >
                  South African
                </option>
                <option
                  value="south korean"
                >
                  South Korean
                </option>
                <option
                  value="spanish"
                >
                  Spanish
                </option>
                <option
                  value="sri lankan"
                >
                  Sri Lankan
                </option>
                <option
                  value="sudanese"
                >
                  Sudanese
                </option>
                <option
                  value="surinamer"
                >
                  Surinamer
                </option>
                <option
                  value="swazi"
                >
                  Swazi
                </option>
                <option
                  value="swedish"
                >
                  Swedish
                </option>
                <option
                  value="swiss"
                >
                  Swiss
                </option>
                <option
                  value="syrian"
                >
                  Syrian
                </option>
                <option
                  value="taiwanese"
                >
                  Taiwanese
                </option>
                <option
                  value="tajik"
                >
                  Tajik
                </option>
                <option
                  value="tanzanian"
                >
                  Tanzanian
                </option>
                <option
                  value="thai"
                >
                  Thai
                </option>
                <option
                  value="togolese"
                >
                  Togolese
                </option>
                <option
                  value="tongan"
                >
                  Tongan
                </option>
                <option
                  value="trinidadian or tobagonian"
                >
                  Trinidadian or Tobagonian
                </option>
                <option
                  value="tunisian"
                >
                  Tunisian
                </option>
                <option
                  value="turkish"
                >
                  Turkish
                </option>
                <option
                  value="tuvaluan"
                >
                  Tuvaluan
                </option>
                <option
                  value="ugandan"
                >
                  Ugandan
                </option>
                <option
                  value="ukrainian"
                >
                  Ukrainian
                </option>
                <option
                  value="uruguayan"
                >
                  Uruguayan
                </option>
                <option
                  value="uzbekistani"
                >
                  Uzbekistani
                </option>
                <option
                  value="venezuelan"
                >
                  Venezuelan
                </option>
                <option
                  value="vietnamese"
                >
                  Vietnamese
                </option>
                <option
                  value="welsh"
                >
                  Welsh
                </option>
                <option
                  value="yemenite"
                >
                  Yemenite
                </option>
                <option
                  value="zambian"
                >
                  Zambian
                </option>
                <option
                  value="zimbabwean"
                >
                  Zimbabwean
                </option>
              </select>
            </NationalityList>
            <label
              htmlFor="race"
            >
              Ethnicity
            </label>
            <select
              id="race"
              name="race"
              required={true}
            >
              <option
                disabled={true}
                value=""
              >
                --select one--
              </option>
              <option
                value="African American"
              >
                African American
              </option>
              <option
                value="Asian"
              >
                Asian
              </option>
              <option
                value="Native American"
              >
                Native American
              </option>
              <option
                value="Latino"
              >
                Latino
              </option>
              <option
                value="Pacific Islander"
              >
                Pacific Islander
              </option>
              <option
                value="Multiracial"
              >
                Multiracial
              </option>
              <option
                value="White"
              >
                White
              </option>
              <option
                value="Other"
              >
                Other
              </option>
            </select>
            <label
              htmlFor="teacher-native-speaker"
            >
              Are You A Native Speaker?
            </label>
            <select
              id="teacher-native-speaker"
              name="native_speaker"
              required={true}
            >
              <option
                value="true"
              >
                Yes
              </option>
              <option
                value="false"
              >
                No
              </option>
            </select>
            <label
              htmlFor="teacher-married"
            >
              Marriage Satus
            </label>
            <select
              defaultValue="no"
              id="teacher-married"
              name="married"
              required={true}
            >
              <option
                disabled={true}
                value=""
              >
                --select one--
              </option>
              <option
                value="Married"
              >
                married
              </option>
              <option
                value="Single"
              >
                single
              </option>
              <option
                value="Domestic-partner"
              >
                domestic partner
              </option>
            </select>
            <label
              htmlFor="teacher-education"
            >
              Education
            </label>
            <select
              id="teacher-education"
              name="highest_degree"
              required={true}
            >
              <option
                disabled="disabled"
                value=""
              >
                -- select one --
              </option>
              <option
                value="No formal education"
              >
                No formal education
              </option>
              <option
                value="Primary education"
              >
                Primary education
              </option>
              <option
                value="Secondary education"
              >
                Secondary education or high school
              </option>
              <option
                value="GED"
              >
                GED
              </option>
              <option
                value="Vocational qualification"
              >
                Vocational qualification
              </option>
              <option
                value="BA"
              >
                Bachelor's degree
              </option>
              <option
                value="Master's degree"
              >
                Master's degree
              </option>
              <option
                value="Doctorate or higher"
              >
                Doctorate or higher
              </option>
            </select>
            <label
              htmlFor="teacher-degree"
            >
              Degree
            </label>
            <input
              id="teacher-degree"
              name="field_of_degree"
              required={true}
              type="text"
            />
            <label
              htmlFor="teacher-school"
            >
              School Attended
            </label>
            <input
              defaultValue="Florida State University"
              id="teacher-school"
              name="school"
              required={true}
              type="text"
            />
            <label
              htmlFor="teacher-certification"
            >
              Certification
            </label>
            <select
              defaultValue="None"
              id="teacher-certification"
              name="certification"
              required={true}
            >
              <option
                value="None"
              >
                None
              </option>
              <option
                value="TEFL"
              >
                TEFL
              </option>
              <option
                value="TESOL"
              >
                TESOL
              </option>
              <option
                value="TEFL/TESOL"
              >
                TEFL/TESOL
              </option>
            </select>
            <label
              htmlFor="teacher-experience"
            >
              Teaching Experience (in years)
            </label>
            <input
              id="teacher-experience"
              min="0"
              name="years_of_experience"
              required={true}
              type="number"
            />
            <label
              htmlFor="teacher-experience-inChina"
            >
              Years Of Experience Teaching In China
            </label>
            <select
              id="teacher-experience-inChina"
              name="years_in_china"
              required={true}
            >
              <option
                value="0"
              >
                0
              </option>
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3+
              </option>
            </select>
            <label
              htmlFor="teacher-experience-notChina"
            >
              Years Of Experience Teaching - Other Countries
            </label>
            <select
              id="teacher-experience-notChina"
              name="years_teaching_abroad"
              required={true}
            >
              <option
                value="0"
              >
                0
              </option>
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3+
              </option>
            </select>
            <button
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </EditTeacherProfile>
    </withRouter(EditTeacherProfile)>
  </Router>
</BrowserRouter>
`;
